data/campaigns/Heir_To_The_Throne/scenarios/17_Scepter_of_Fire.cfg

```diff
@@ -617,6 +617,51 @@
         {CLEAR_VARIABLE delfador_moves}
     [/event]
 
+    # Introductory dialog to give the player a hint that she may lose units to
+    # {EXPAND_LAVA}.
+    [event]
+        name=turn 2
+        # The noise was played by {EXPAND_LAVA}.
+        [message]
+            id=Konrad
+            message=_ "That noise was made by no troll! Can you tell its source?"
+        [/message]
+        [if]
+            [have_unit]
+                race=dwarf
+            [/have_unit]
+            [then]
+                [message]
+                    race=dwarf
+                    message=_ "Lord Konrad, yon rumble was made by nae enemy o’ us. ’Tis the echoes of the very bedrock witherin’ and crumbling beneath us that we ’eard. Are ye nae feeling the heat redoubling? Is yer breath nae difficult for the air, sowled by ashes? I beseech that we not tarry ’ere any more than we ha’ to, lest the forming abysses claim us."
+                [/message]
+            [/then]
+            [elseif]
+                # At this point, the lava is shrouded. Let's assume that among
+                # mages, only high-level ones know the rumble was caused by
+                # lava, either because they know the Sceptre's lore or because
+                # they are can sense the fire. Remember that level 1 Mages are
+                # apprentices, according to lore.
+                [have_unit]
+                    type=Arch Mage,Great Mage,Silver Mage
```

Comment:
The comment suggests that `type=Mage of Light` should also be accepted; however limiting it to Red Mages' advancements makes sense for the knowledge of fire.